From c17f3aa64a264a5fec7d3c5f8d1e9415b60a55b4 Mon Sep 17 00:00:00 2001
From: Alynx Zhou <alynx.zhou@gmail.com>
Date: Wed, 15 May 2024 10:09:09 +0800
Subject: [PATCH] inputMethod: Reset preedit cursor when preedit text is
 cleared

The preedit cursor position should be 0 when there is no preedit text,
currently it will pass 1, which is the wrong previous value.
---
 js/misc/inputMethod.js | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/js/misc/inputMethod.js b/js/misc/inputMethod.js
index d8c3e62f0c7..2642aca9e78 100644
--- a/js/misc/inputMethod.js
+++ b/js/misc/inputMethod.js
@@ -163,9 +163,7 @@ export const InputMethod = GObject.registerClass({
     }
 
     _onHidePreeditText() {
-        this.set_preedit_text(
-            null, this._preeditPos, this._preeditAnchor,
-            this._preeditCommitMode);
+        this.set_preedit_text(null, 0, 0, this._preeditCommitMode);
         this._preeditVisible = false;
     }
 
-- 
GitLab

